File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.java
import java.math.BigInteger;
class Solution {
    public int numSteps(String s) {
          BigInteger ss = new BigInteger(s, 2);
        int c = 0;
        while (!ss.equals(BigInteger.ONE)) {
            if (ss.mod(BigInteger.valueOf(2)).equals(BigInteger.ONE)) {
                ss = ss.add(BigInteger.ONE);
            } else {
                ss = ss.divide(BigInteger.valueOf(2));
            }
            c++;
        }
        return c;
    }
}

File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.py
class Solution:
    def numSteps(self, s: str) -> int:
        ss=int(s,2)
        c=0
        while ss!=1:
            if ss % 2 != 0:
                ss=ss+1
                c+=1
            elif ss % 2 == 0:
                ss=ss//2
                c+=1
        return c